rust/src/transformations/make_stable_expr/expr_binary/test.rs

```diff
@@ -421,3 +445,23 @@ fn test_floor_div() -> Fallible<()> {
     test_binary!(get_i32_data, floor_div, [None, Some(1), None])?;
     Ok(())
 }
+
+#[test]
+fn test_rem() -> Fallible<()> {
+    test_binary!(
+        get_f64_data,
+        rem,
+        [
+            Some(f64::NAN),
+            Some(f64::NAN),
+            Some(f64::NAN),
+            Some(f64::NAN),
+            None,
+            None,
+            Some(f64::NAN),
+            Some(f64::NAN)
+        ]
+    )?;
+    test_binary!(get_i32_data, rem, [None, Some(0), None])?;
+    Ok(())
+}
```

Comment:
Not something to block this PR, and there's a longer note in an earlier, but I'm not comfortable with this style of testing.

Since the inputs are separated from the asserted outputs, it's easy to make this just an assertion of the current behavior, without thinking about what that behavior should be. I believe you're being careful, but I'm not really checking that each assertion makes sense, and it may also be hard for a future maintainer.